be/src/util/system_metrics.cpp

```diff
@@ -135,6 +143,9 @@ SystemMetrics::~SystemMetrics() {
     if (_line_ptr != nullptr) {
         free(_line_ptr);
     }
+    for (auto& it : _runtime_filter_metrics) {
+        delete it.second;
```

Comment:
may consider using std::unique_ptr to manage the pointers instead of using the raw one. so the resource deallocation will be as easy as  `_runtime_filter_metrics.clear()`